src/day8.scala

```diff
@@ -0,0 +1,130 @@
+// using scala 3.0.2
+
+package day8
+
+import scala.util.Using
+import scala.io.Source
+
+import Segment.*
+import Digit.*
+
+@main def part1: Unit =
+  println(s"The solution is ${part1(readInput())}")
+
+@main def part2: Unit =
+  println(s"The solution is ${part2(readInput())}")
+
+def readInput(): String =
+  Using.resource(Source.fromFile("input/day8"))(_.mkString)
+
+type Segments = Seq[Segment]
+
+def part1(input: String) =
+
+  def getDisplay(line: String) = line.split("[|]")(1).trim
+
+  val uniqueDigits =
+    for
+      display <- input.linesIterator.map(getDisplay)
+      segments <- display.split(" ")
+      uniqueDigit <- parseUniqueDigit(segments)
+    yield
+      uniqueDigit
+
+  uniqueDigits.length
+end part1
+
+def part2(input: String) =
```

Comment:
I would prefer to see the implementation details of the part 1 (the method `parseUniqueDigit`) rather than having to scroll over the `part2`.

Reply:
Also, the return type would be welcome

Reply:
I moved `parseUniqueDigit` into `part1`